### PR TITLE
[BEAM-1122] Change Dataflow profiling option to saveProfilesToGcs

### PR DESCRIPTION
**Pull request by bjchambers**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [*] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [*] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [*] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [*] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by bjchambers** [comment]
R: @davorbonaci

File: runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/options/DataflowProfilingOptions.java [hunk 1]
```diff
@@ -30,10 +30,10 @@
 @Hidden
 public interface DataflowProfilingOptions {
 
-  @Description("Whether to periodically dump profiling information to local disk.\n"
-      + "WARNING: Enabling this option may fill local disk with profiling information.")
-  boolean getEnableProfilingAgent();
-  void setEnableProfilingAgent(boolean enabled);
+  @Description("When set to a non-empty value, enables recording profiles and saving them to GCS.\n"
```
Comment (davorbonaci) [inline runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/options/DataflowProfilingOptions.java]: JIRA issue, with a non-backward-compatible label.

File: runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowRunner.java [hunk 2]
```diff
@@ -262,6 +263,10 @@ public static DataflowRunner fromOptions(PipelineOptions options) {
         "DataflowRunner requires stagingLocation, and it is missing in PipelineOptions.");
     validator.validateOutputFilePrefixSupported(dataflowOptions.getStagingLocation());
 
+    if (!Strings.isNullOrEmpty(dataflowOptions.getSaveProfilesToGcs())) {
```
Comment (davorbonaci) [inline runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowRunner.java]: is there an option to disable validation?

(conceptually, the user might not have access, while the VMs do have)

File: runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/options/DataflowProfilingOptions.java (hunk 1 above)
Comment (bjchambers) [inline runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/options/DataflowProfilingOptions.java]: https://issues.apache.org/jira/browse/BEAM-1122

File: runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowRunner.java (hunk 2 above)
Comment (bjchambers) [inline runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowRunner.java]: Not currently. We may add one in the future, but requiring that the user (and VMs) have access seems like the simplest path for now.

**Comment by davorbonaci** [comment]
LGTM. Merging.

**Pull request closed without merging**